edx_rbac/utils.py

```diff
@@ -62,8 +63,12 @@ def user_has_access_via_database(user, role_name, role_assignment_class, context
         return False
 
     if context:
-        return role_assignment.get_context() in (context, ALL_ACCESS_CONTEXT)
-
+        context_in_database = role_assignment.get_context()
+        if isinstance(context_in_database, string_types):
+            return context_in_database in (context, ALL_ACCESS_CONTEXT)
+        else:
+            context_in_database.append(ALL_ACCESS_CONTEXT)
```

Comment:
so, i don't think this will work in the case where the user has a context like `['a', 'b', '*']` and the context passed in is `c`. in this case, we would erroneously deny access.

Reply:
Thanks for catching this. The ALL_ACCESS_CONTEXT is redundant in '''else''' case since it is already handled in the '''if''' case.